src/category_theory/limits/shapes/images.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2019 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison, Markus Himmel
+-/
+import category_theory.limits.shapes.equalizers
+
+/-!
+# Categorical images
+
+We define the categorical image of `f` as a factorisation `f = e ≫ m` through a monomorphism `m`,
+so that `m` factors through the `m'` in any other such factorisation.
+
+## Main statements
+
+* When `C` has equalizers, the morphism `m` is an epimorphism.
+
+## Future work
+* TODO: coimages, and abelian categories.
+* TODO: connect this with existing working in the group theory and ring theory libraries.
+
+-/
+
+universes v u
+
+open category_theory
+open category_theory.limits.walking_parallel_pair
+
+namespace category_theory.limits
+
+variables {C : Type u} [𝒞 : category.{v} C]
+include 𝒞
+
+variables {X Y : C} (f : X ⟶ Y)
+
+/-- A factorisation of a morphism `f = e ≫ m`, with `m` monic. -/
+-- FIXME I think there is a bug in the linter. Why should you expect an inhabited instance here?
```

Comment:
I think the reasoning is as follows: If you define a type `foo` that depends on some `x` and `y`, then you should be able to provide at least one example of `x` and `y` for which `foo x y` is inhabited, otherwise there is no point in defining the type.
I'm not sure if I agree with this.
For example, in this case you could show that `mono_factorisation id` is "canonically" inhabited. I haven't read the following yet, but you probably *are* going to do something like that and prove some simp-lemmas about it.
So maybe such an instance is useful... Otoh, I'm not sure how we will later use that it's such an instance. Instead of just having a `def` for `mono_factorisation id`.
@gebner Could you enlighten us?

Reply:
Ok! I will add `def self [mono f] : mono_factorisation f :=`, note this as an inhabited instance to satisfy the linter, and use it to prove the image of a monic is isomorphic to the source.